motor/discovery/gcp/mql_asset_objects.go

```diff
@@ -230,3 +313,39 @@ func bigQueryDatasets(m *MqlDiscovery, project string, tc *providers.Config) []*
 	}
 	return assets
 }
+
+func mapInstanceStatus(state string) asset.State {
+	switch state {
+	case "RUNNING":
+		return asset.State_STATE_RUNNING
+	case "PROVISIONING":
+		return asset.State_STATE_PENDING
+	case "STAGING":
+		return asset.State_STATE_PENDING
+	case "STOPPED":
+		return asset.State_STATE_STOPPED
+	case "STOPPING":
+		return asset.State_STATE_STOPPING
+	case "SUSPENDED":
+		return asset.State_STATE_STOPPED
+	case "SUSPENDING":
+		return asset.State_STATE_STOPPING
+	case "TERMINATED":
+		return asset.State_STATE_TERMINATED
+	default:
+		log.Warn().Str("state", state).Msg("unknown gcp instance state")
+		return asset.State_STATE_UNKNOWN
+	}
+}
+
+func disksContainWindows(disks []interface{}) bool {
+	for _, d := range disks {
+		feats := d.(map[string]interface{})["guestOsFeatures"].([]interface{})
```

Comment:
this kind of stuff is pretty nasty. Especially, if we need to be checking some deeply nested structure. I guess it's rarely needed on an asset level but still good to improve at some point

Reply:
This is fixed with #891 When that gets merged the problem will be gone